src/ssl.mli

```diff
@@ -209,9 +209,18 @@ type protocol =
   | SSLv23 (** accept all possible protocols (SSLv2 if supported by openssl,
                SSLv3, TLSv1, TLSv1.1, TLSv1.2, and TLSv1.3) *)
   | SSLv3 (** only SSL v3 protocol *)
-  | TLSv1 (** only TLS v1 protocol *)
-  | TLSv1_1 (** only TLS v1.1 protocol *)
-  | TLSv1_2 (** only TLS v1.2 protocol *)
+  | TLSv1
+    [@ocaml.alert deprecated
+      "TLSv1 is officialy deprecated, use a more recent one"]
+
+  | TLSv1_1
+    [@ocaml.alert deprecated
+      "TLSv1.1 is officialy deprecated, use a more recent one"]
+
+  | TLSv1_2
```

Comment:
TLS v1.2 isn't deprecated https://en.wikipedia.org/wiki/Transport_Layer_Security#TLS_1.2

Reply:
Fixed this. I was in advance ;-)

Reply:
I like this PR, because now ocaml-ssl compiles without warning from GCC

Reply:
Sure, but what about we use non-deprecated APIs instead, like #59 proposed?